### ISSUE TITLE
Sticky Shift partially working

### ISSUE THREAD

**Issue by SchattengestaIt** [tdm/android_device_fxtec_pro1#58]
Sticky shift doesn´t work on certain text fields like password input and the text editors of the Fxtec forum or Github. I remember this not being a problem with previous keyboard phones so it should theoretically be fixable.

**Comment by Gigadoc2** [comment]
I think that the app controls whether those features (stick shift, also symbol picker and autocorrect) are enabled for a given input. For the password fields this seems deliberate, I can understand why you'd like to have sticky shift there, but most people probably would be confused by automatic changes happening on a field which they can't see.

For the text fields, this seems like a decision the browser made? Though there I have no idea why they would want to disable that.

But the bottom line is, I don't think this can be fixed on the LIineageOS side, unless sticky shift is also moved into the kernel module.

**Comment by enigma9o7** [comment]
I wish I didnt have to agree, but yeah I think this is an AOSP or app probem, this happens on photon q stock or cm/los too.  Some fields the sticky shift just doesn;t work.  When you get used to using it, it's annoying when it doesnt work as expected for example @ symbol, etc.  If there is a way to make it work always, that would be much better but probably not a simple fix.

I see how that could be an issue with passwords especially since you don't see that the shift didn't stick; however when you retype the password it won't stick the next time either and you'll never know ;)  unless password field requires a capitol or symbol or something and you're failing cuz of that.  Or you login from your computer.

**Comment by SchattengestaIt** [comment]
The keyboard management is generally strange with the FxTec/LineageOS.
You have behaviour like double space for an added dot ( **.** ) and dots meaning that the next letter will be a capital one. There doesn't seem to be any way to disable that. Long press gives out menus that don't pop up with other systems so there has to be some configuration from the software side which most likely can be changed.

I've also noticed a software keyboard, that some text fields (most likely in flash games) have posted different keys. It seems like different config files are used in both cases but I didn't have had problems with Gmail and browsers so far.
Passwords aren't even that big of a problem, but I have to type every longer message on a note app in order to have sticky shift. And that can be time consuming and inconvenient.


The mentioned problems (like sticky shift not working in some cases) doesn't happen with my Droid 4 although I wasn't able to try every app since the phone is damn slow/unusuable. Could be that they changed the AOSP since then but it would still be strange. It is also possible that I still have my custom keyboard on that device which has a different behaviour.

I'll keep looking into this and if I ever find a solution I'll tell you. There has to be a way to solve it.

**Comment by SchattengestaIt** [comment]
Or better said: I have disabled things like double spacebacebar to insert a period ( . ) in both AOSP as well as my software keyboard (Clavis). That doesn't change anything though.
There simply isn't a setting for the physical keyboard or I can't find it. (probably just some random file laying somewhere in the system folders).

**Comment by enigma9o7** [comment]
That is true.  The settings are missing.  I have posted that as a seperate issue here:

https://github.com/tdm/android_device_fxtec_pro1/issues/53
